Add Optimize backup functionality when running 8.6 against an OpenSearch DB

Engineering DRI: @oleksandr-kriuchenko-lohika

### Context

<!-- Please, describe the context and need for this task.-->

With [extended OpenSearch support](https://github.com/camunda/product-hub/issues/1908), we did not implement support for backups. This was added later in the [full support](https://github.com/camunda/product-hub/issues/2473), and will be released in 8.8. However, the backups should be considered part of the critical functionality of Optimize, and should thus be supported in the 8.6 release.

We need to implement the missing functionality, and release this as part of an 8.6 patch. We should also ensure that this is part of the upcoming 8.7 release.

The functionality is currently not working, although the Optimize API is available. The error is logged as:

```
10:53:25.303 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeIncidentImportMediator - Records of type INCIDENT from partition 2 imported in page: 0

10:53:25.307 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-user-task found to read records from!

10:53:25.307 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeUserTaskImportMediator - Records of type USER_TASK from partition 2 imported in page: 0

10:53:25.315 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-variable found to read records from!

10:53:25.315 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeVariableImportMediator - Records of type VARIABLE from partition 2 imported in page: 0

10:53:25.320 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-process found to read records from!

10:53:25.320 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeProcessDefinitionImportMediator - Records of type PROCESS from partition 3 imported in page: 0

10:53:25.325 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-process-instance found to read records from!

10:53:25.325 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeProcessInstanceImportMediator - Records of type PROCESS_INSTANCE from partition 3 imported in page: 0

10:53:25.332 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-incident found to read records from!

10:53:25.332 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeIncidentImportMediator - Records of type INCIDENT from partition 3 imported in page: 0

10:53:25.338 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-user-task found to read records from!

10:53:25.338 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeUserTaskImportMediator - Records of type USER_TASK from partition 3 imported in page: 0

10:53:25.343 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-variable found to read records from!

10:53:25.343 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeVariableImportMediator - Records of type VARIABLE from partition 3 imported in page: 0

10:53:38.560 [qtp1387026407-81] INFO  i.c.optimize.service.BackupService - Triggering backup with ID 1122334455

10:53:38.581 [Thread-48] INFO  i.c.o.s.d.r.es.SnapshotRepositoryOS - Triggering async snapshot camunda_optimize_1122334455_8.6.4_part_1_of_2.

10:53:38.625 [Thread-48] INFO  i.c.o.s.d.r.es.SnapshotRepositoryOS - Triggering async snapshot camunda_optimize_1122334455_8.6.4_part_2_of_2.

10:53:42.214 [aws-java-sdk-NettyEventLoop-1-0] ERROR i.c.o.s.d.r.es.SnapshotRepositoryOS - Failed to take snapshot [camunda_optimize_1122334455_8.6.4_part_1_of_2]

java.util.concurrent.CompletionException: org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [snapshot_exception] [optimize-backup:camunda_optimize_1122334455_8.6.4_part_1_of_2/f1xIx6VYSqu1SMbLpkB98w] failed to update snapshot in repository

    at java.base/java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:368)

    at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1189)

    at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2341)

    at org.opensearch.client.transport.aws.AwsSdk2Transport.lambda$executeAsync$9(AwsSdk2Transport.java:433)

    at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)

    at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)

    at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)

    at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler.finalizeResponse(ResponseHandler.java:198)

    at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler.access$900(ResponseHandler.java:77)

    at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler$PublisherAdapter$1.onComplete(ResponseHandler.java:379)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.publishMessage(HandlerPublisher.java:402)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.flushBuffer(HandlerPublisher.java:338)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.receivedDemand(HandlerPublisher.java:291)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.access$200(HandlerPublisher.java:61)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher$ChannelSubscription$1.run(HandlerPublisher.java:495)

    at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)

    at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)

    at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)

    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:566)

    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)

    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)

    at java.base/java.lang.Thread.run(Thread.java:1583)

Caused by: org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [snapshot_exception] [optimize-backup:camunda_optimize_1122334455_8.6.4_part_1_of_2/f1xIx6VYSqu1SMbLpkB98w] failed to update snapshot in repository

    at org.opensearch.client.transport.aws.AwsSdk2Transport.parseResponse(AwsSdk2Transport.java:540)

    at org.opensearch.client.transport.aws.AwsSdk2Transport.lambda$executeAsync$8(AwsSdk2Transport.java:438)

    at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)

    ... 20 common frames omitted

10:53:42.472 [aws-java-sdk-NettyEventLoop-1-1] ERROR i.c.o.s.d.r.es.SnapshotRepositoryOS - Failed to take snapshot [camunda_optimize_1122334455_8.6.4_part_2_of_2]

java.util.concurrent.CompletionException: org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [snapshot_exception] [optimize-backup:camunda_optimize_1122334455_8.6.4_part_2_of_2/RAkPR4h0Tc2n89rqAYkEqg] failed to update snapshot in repository

    at java.base/java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:368)

    at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1189)

    at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2341)

    at org.opensearch.client.transport.aws.AwsSdk2Transport.lambda$executeAsync$9(AwsSdk2Transport.java:433)

    at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)

    at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)

    at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)

    at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler.finalizeResponse(ResponseHandler.java:198)

    at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler.access$900(ResponseHandler.java:77)

    at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler$PublisherAdapter$1.onComplete(ResponseHandler.java:379)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.publishMessage(HandlerPublisher.java:402)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.flushBuffer(HandlerPublisher.java:338)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.receivedDemand(HandlerPublisher.java:291)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.access$200(HandlerPublisher.java:61)

    at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher$ChannelSubscription$1.run(HandlerPublisher.java:495)

    at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)

    at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)

    at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)

    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:566)

    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)

    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)

    at java.base/java.lang.Thread.run(Thread.java:1583)

Caused by: org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [snapshot_exception] [optimize-backup:camunda_optimize_1122334455_8.6.4_part_2_of_2/RAkPR4h0Tc2n89rqAYkEqg] failed to update snapshot in repository

    at org.opensearch.client.transport.aws.AwsSdk2Transport.parseResponse(AwsSdk2Transport.java:540)

    at org.opensearch.client.transport.aws.AwsSdk2Transport.lambda$executeAsync$8(AwsSdk2Transport.java:438)

    at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)

    ... 20 common frames omitted

10:53:49.132 [qtp1387026407-81] WARN  o.e.j.ee10.servlet.ServletChannel - /actuator/backups

jakarta.servlet.ServletException: Request processing failed: java.lang.NullPointerException: Cannot invoke "java.util.Map.entrySet()" because the return value of "io.camunda.optimize.service.db.reader.BackupReader.getAllOptimizeSnapshotsByBackupId()" is null

    at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1022)

    at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903)

    at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:527)

    at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885)

    at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:614)

    at org.eclipse.jetty.ee10.servlet.ServletHolder.handle(ServletHolder.java:736)

    at org.eclipse.jetty.ee10.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1614)

    at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:108)

    at org.springframework.security.web.FilterChainProxy.lambda$doFilterInternal$3(FilterChainProxy.java:231)

–

10:53:55.311 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeIncidentImportMediator - Records of type INCIDENT from partition 2 imported in page: 0

10:53:55.316 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-user-task found to read records from!

10:53:55.316 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeUserTaskImportMediator - Records of type USER_TASK from partition 2 imported in page: 0

10:53:55.322 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-variable found to read records from!

10:53:55.322 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeVariableImportMediator - Records of type VARIABLE from partition 2 imported in page: 0

10:53:55.327 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-process found to read records from!

10:53:55.327 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeProcessDefinitionImportMediator - Records of type PROCESS from partition 3 imported in page: 0

10:53:55.332 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-process-instance found to read records from!

10:53:55.332 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeProcessInstanceImportMediator - Records of type PROCESS_INSTANCE from partition 3 imported in page: 0

10:53:55.338 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-incident found to read records from!

10:53:55.338 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeIncidentImportMediator - Records of type INCIDENT from partition 3 imported in page: 0

10:53:55.343 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-user-task found to read records from!

10:53:55.343 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeUserTaskImportMediator - Records of type USER_TASK from partition 3 imported in page: 0

10:53:55.349 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-variable found to read records from!

10:53:55.349 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeVariableImportMediator - Records of type VARIABLE from partition 3 imported in page: 0

10:54:20.892 [qtp1387026407-81] ERROR i.c.optimize.service.BackupService - No Optimize backup with ID [1122334455] could be found.

10:54:24.831 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-process found to read records from!

10:54:24.831 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeProcessDefinitionImportMediator - Records of type PROCESS from partition 1 imported in page: 0

10:54:24.954 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-process-instance found to read records from!

10:54:24.954 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeProcessInstanceImportMediator - Records of type PROCESS_INSTANCE from partition 1 imported in page: 0

10:54:25.291 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-incident found to read records from!

10:54:25.291 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeIncidentImportMediator - Records of type INCIDENT from partition 1 imported in page: 0

10:54:25.297 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-user-task found to read records from!

10:54:25.297 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeUserTaskImportMediator - Records of type USER_TASK from partition 1 imported in page: 0

10:54:25.303 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-variable found to read records from!

10:54:25.303 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeVariableImportMediator - Records of type VARIABLE from partition 1 imported in page: 0

10:54:25.310 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-process found to read records from!

10:54:25.310 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeProcessDefinitionImportMediator - Records of type PROCESS from partition 2 imported in page: 0

10:54:25.315 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-process-instance found to read records from!

10:54:25.315 [ZeebeImportScheduler-1] INFO  i.c.o.s.i.z.m.ZeebeProcessInstanceImportMediator - Records of type PROCESS_INSTANCE from partition 2 imported in page: 0

10:54:25.320 [ZeebeImportScheduler-1] WARN  i.c.o.s.i.z.f.AbstractZeebeRecordFetcher - No Zeebe index with alias zeebe-record-incident found to read records from!
```

### Acceptance Criteria

<!-- Please list the Acceptance Criteria. It will be used during Review -->

### Testing Notes

Requires testing for all the setup combinations:
- ES, OS
- SM, SaaS
- ES/OS containers vs AWS services

### Links

<!-- Add links to related issues -->

https://jira.camunda.com/browse/SUPPORT-25836

### Breakdown

<!-- A breakdown of tasks that need to be completed in order for this to be ready for review. -->
<!--
- [ ] #123
- [ ] Step X
-->

```[tasklist]
### Pull Requests
```

## Task Lifecycle

For managing the issue lifecycle, please use the workflow commands. You can see the available
commands by writing `/help` as a comment on this issue.

### Review

#### Engineering Review

- [x] All code targeting the main branch has been reviewed by at least one Engineer
- [x] The PR targeting the main branch has been approved by the reviewing Engineer
- [x] If the API has changed, the API documentation is updated
- [x] All other PRs (docs, controller etc.) in the breakdown have been approved

#### QA Review

- [ ] The change is implemented as described on all target environments/versions/modes
- [ ] The documentation changes are as expected

### Completion

- [ ] All Review stages are successfully completed
- [ ] All associated PRs are merged to the main branch(es) and stable branches
- [ ] The correct version labels are applied to the issue

--- a/optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/DatabaseIntegrationTestExtension.java
+++ b/optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/DatabaseIntegrationTestExtension.java
@@ -340,6 +340,12 @@ public class DatabaseIntegrationTestExtension implements BeforeEachCallback, Aft
     databaseTestService.createMissingIndices(indexMappingCreatorBuilder, aliases, aKey);
   }
 
+  public String[] getImportIndices() {
+    return databaseTestService.getImportIndices().stream()
+        .map(getIndexNameService()::getOptimizeIndexAliasForIndex)
+        .toArray(String[]::new);
+  }
+
   public void setActivityStartDatesToNull(final String processDefinitionKey) {
     final ScriptData scriptData =
         new ScriptData(

--- a/optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/DatabaseTestService.java
+++ b/optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/DatabaseTestService.java
@@ -193,6 +193,8 @@ public abstract class DatabaseTestService {
 
   public abstract void cleanSnapshots(final String snapshotRepositoryName);
 
+  public abstract List<String> getImportIndices();
+
   protected abstract <T extends OptimizeDto> List<T> getInstancesById(
       final String indexName,
       final List<String> instanceIds,

--- a/optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/ElasticsearchDatabaseTestService.java
+++ b/optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/ElasticsearchDatabaseTestService.java
@@ -87,6 +87,7 @@ import io.camunda.optimize.service.db.es.builders.OptimizeUpdateRequestBuilderES
 import io.camunda.optimize.service.db.es.reader.ElasticsearchReaderUtil;
 import io.camunda.optimize.service.db.es.schema.ElasticSearchIndexSettingsBuilder;
 import io.camunda.optimize.service.db.es.schema.ElasticSearchMetadataService;
+import io.camunda.optimize.service.db.es.schema.ElasticSearchSchemaManager;
 import io.camunda.optimize.service.db.es.schema.index.ExternalProcessVariableIndexES;
 import io.camunda.optimize.service.db.es.schema.index.ProcessInstanceIndexES;
 import io.camunda.optimize.service.db.es.schema.index.TerminatedUserSessionIndexES;
@@ -545,39 +546,6 @@ public class ElasticsearchDatabaseTestService extends DatabaseTestService {
   }
 
   @Override
-  public void createRepoSnapshot(final String snapshotRepositoryName) {
-    try {
-      getOptimizeElasticClient()
-          .getEsClient()
-          .snapshot()
-          .createRepository(
-              CreateRepositoryRequest.of(
-                  b ->
-                      b.name(snapshotRepositoryName)
-                          .repository(r -> r.fs(s -> s.settings(t -> t.location("/var/tmp"))))));
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-  }
-
-  @Override
-  public void cleanSnapshots(String snapshotRepositoryName) {
-    try {
-      getOptimizeElasticClient()
-          .getEsClient()
-          .snapshot()
-          .delete(
-              DeleteSnapshotRequest.of(b -> b.repository(snapshotRepositoryName).snapshot("*")));
-      getOptimizeElasticClient()
-          .getEsClient()
-          .snapshot()
-          .deleteRepository(DeleteRepositoryRequest.of(b -> b.name(snapshotRepositoryName)));
-    } catch (Exception e) {
-      log.warn("Delete failed, no snapshots to delete from repository {}", snapshotRepositoryName);
-    }
-  }
-
-  @Override
   @SneakyThrows
   public boolean indexExistsCheckWithoutApplyingOptimizePrefix(final String indexName) {
     final OptimizeElasticsearchClient esClient = getOptimizeElasticClient();
@@ -655,6 +623,67 @@ public class ElasticsearchDatabaseTestService extends DatabaseTestService {
   @Override
   public DatabaseType getDatabaseVendor() {
     return DatabaseType.ELASTICSEARCH;
+  }
+
+  @Override
+  public void createSnapshot(
+      final String snapshotRepositoryName, final String snapshotName, final String[] indexNames) {
+    final CreateSnapshotRequest createSnapshotRequest =
+        CreateSnapshotRequest.of(
+            b ->
+                b.repository(snapshotRepositoryName)
+                    .snapshot(snapshotName)
+                    .indices(Arrays.stream(indexNames).toList())
+                    .includeGlobalState(false)
+                    .waitForCompletion(true));
+    try {
+      getOptimizeElasticClient()
+          .triggerSnapshotAsync(createSnapshotRequest)
+          .get(10, TimeUnit.SECONDS);
+    } catch (final InterruptedException | ExecutionException | TimeoutException e) {
+      throw new OptimizeRuntimeException("Exception during creation snapshot:", e);
+    }
+  }
+
+  @Override
+  public void createRepoSnapshot(final String snapshotRepositoryName) {
+    try {
+      getOptimizeElasticClient()
+          .getEsClient()
+          .snapshot()
+          .createRepository(
+              CreateRepositoryRequest.of(
+                  b ->
+                      b.name(snapshotRepositoryName)
+                          .repository(r -> r.fs(s -> s.settings(t -> t.location("/var/tmp"))))));
+    } catch (final IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void cleanSnapshots(final String snapshotRepositoryName) {
+    try {
+      getOptimizeElasticClient()
+          .getEsClient()
+          .snapshot()
+          .delete(
+              DeleteSnapshotRequest.of(b -> b.repository(snapshotRepositoryName).snapshot("*")));
+      getOptimizeElasticClient()
+          .getEsClient()
+          .snapshot()
+          .deleteRepository(DeleteRepositoryRequest.of(b -> b.name(snapshotRepositoryName)));
+    } catch (final Exception e) {
+      log.warn("Delete failed, no snapshots to delete from repository {}", snapshotRepositoryName);
+    }
+  }
+
+  @Override
+  public List<String> getImportIndices() {
+    return ElasticSearchSchemaManager.getAllNonDynamicMappings().stream()
+        .filter(IndexMappingCreator::isImportIndex)
+        .map(IndexMappingCreator::getIndexName)
+        .toList();
   }
 
   @Override
@@ -1062,26 +1091,6 @@ public class ElasticsearchDatabaseTestService extends DatabaseTestService {
       return esClient.elasticsearchClient().count(countRequest).count();
     } catch (final IOException e) {
       throw new OptimizeIntegrationTestException(e);
-    }
-  }
-
-  @Override
-  public void createSnapshot(
-      final String snapshotRepositoryName, final String snapshotName, final String[] indexNames) {
-    CreateSnapshotRequest createSnapshotRequest =
-        CreateSnapshotRequest.of(
-            b ->
-                b.repository(snapshotRepositoryName)
-                    .snapshot(snapshotName)
-                    .indices(Arrays.stream(indexNames).toList())
-                    .includeGlobalState(false)
-                    .waitForCompletion(true));
-    try {
-      getOptimizeElasticClient()
-          .triggerSnapshotAsync(createSnapshotRequest)
-          .get(10, TimeUnit.SECONDS);
-    } catch (InterruptedException | ExecutionException | TimeoutException e) {
-      throw new OptimizeRuntimeException("Exception during creation snapshot:", e);
     }
   }
 

--- a/optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/OpenSearchDatabaseTestService.java
+++ b/optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/OpenSearchDatabaseTestService.java
@@ -39,6 +39,7 @@ import io.camunda.optimize.service.db.os.externalcode.client.dsl.QueryDSL;
 import io.camunda.optimize.service.db.os.externalcode.client.dsl.RequestDSL;
 import io.camunda.optimize.service.db.os.schema.OpenSearchIndexSettingsBuilder;
 import io.camunda.optimize.service.db.os.schema.OpenSearchMetadataService;
+import io.camunda.optimize.service.db.os.schema.OpenSearchSchemaManager;
 import io.camunda.optimize.service.db.os.schema.index.ExternalProcessVariableIndexOS;
 import io.camunda.optimize.service.db.os.schema.index.ProcessInstanceIndexOS;
 import io.camunda.optimize.service.db.os.schema.index.TerminatedUserSessionIndexOS;
@@ -66,12 +67,16 @@ import io.camunda.zeebe.protocol.record.value.BpmnElementType;
 import java.io.IOException;
 import java.time.OffsetDateTime;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 import lombok.SneakyThrows;
@@ -114,6 +119,10 @@ import org.opensearch.client.opensearch.indices.IndexSettings;
 import org.opensearch.client.opensearch.indices.PutIndicesSettingsRequest;
 import org.opensearch.client.opensearch.indices.RefreshRequest;
 import org.opensearch.client.opensearch.indices.get_alias.IndexAliases;
+import org.opensearch.client.opensearch.snapshot.CreateRepositoryRequest;
+import org.opensearch.client.opensearch.snapshot.CreateSnapshotRequest;
+import org.opensearch.client.opensearch.snapshot.DeleteRepositoryRequest;
+import org.opensearch.client.opensearch.snapshot.DeleteSnapshotRequest;
 
 @Slf4j
 public class OpenSearchDatabaseTestService extends DatabaseTestService {
@@ -616,7 +625,29 @@ public class OpenSearchDatabaseTestService extends DatabaseTestService {
   @Override
   public void createSnapshot(
       String snapshotRepositoryName, String snapshotName, String[] indexNames) {
-    // not implemented
+    CreateSnapshotRequest createSnapshotRequest =
+        CreateSnapshotRequest.of(
+            b ->
+                b.repository(snapshotRepositoryName)
+                    .snapshot(snapshotName)
+                    .indices(Arrays.stream(indexNames).toList())
+                    .includeGlobalState(false)
+                    .waitForCompletion(true));
+    try {
+      getOptimizeOpenSearchClient()
+          .triggerSnapshotAsync(createSnapshotRequest)
+          .get(10, TimeUnit.SECONDS);
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      throw new OptimizeRuntimeException("Exception during creation snapshot:", e);
+    }
+  }
+
+  @Override
+  public List<String> getImportIndices() {
+    return OpenSearchSchemaManager.getAllNonDynamicMappings().stream()
+        .filter(IndexMappingCreator::isImportIndex)
+        .map(IndexMappingCreator::getIndexName)
+        .toList();
   }
 
   @Override
@@ -728,12 +759,36 @@ public class OpenSearchDatabaseTestService extends DatabaseTestService {
 
   @Override
   public void createRepoSnapshot(String snapshotRepositoryName) {
-    // not implemented
+    try {
+      getOptimizeOpenSearchClient()
+          .getOpenSearchClient()
+          .snapshot()
+          .createRepository(
+              CreateRepositoryRequest.of(
+                  b ->
+                      b.name(snapshotRepositoryName)
+                          .settings(s -> s.location("/var/tmp"))
+                          .type("fs")));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
   }
 
   @Override
   public void cleanSnapshots(String snapshotRepositoryName) {
-    // not implemented
+    try {
+      getOptimizeOpenSearchClient()
+          .getOpenSearchClient()
+          .snapshot()
+          .delete(
+              DeleteSnapshotRequest.of(b -> b.repository(snapshotRepositoryName).snapshot("*")));
+      getOptimizeOpenSearchClient()
+          .getOpenSearchClient()
+          .snapshot()
+          .deleteRepository(DeleteRepositoryRequest.of(b -> b.name(snapshotRepositoryName)));
+    } catch (Exception e) {
+      log.warn("Delete failed, no snapshots to delete from repository {}", snapshotRepositoryName);
+    }
   }
 
   @Override

--- a/optimize/backend/src/main/java/io/camunda/optimize/service/BackupService.java
+++ b/optimize/backend/src/main/java/io/camunda/optimize/service/BackupService.java
@@ -15,26 +15,20 @@ import static io.camunda.optimize.dto.optimize.rest.SnapshotState.SUCCESS;
 import static java.util.stream.Collectors.groupingBy;
 import static java.util.stream.Collectors.joining;
 
-import co.elastic.clients.elasticsearch._types.ShardFailure;
-import co.elastic.clients.elasticsearch.snapshot.SnapshotInfo;
 import io.camunda.optimize.dto.optimize.BackupState;
 import io.camunda.optimize.dto.optimize.rest.BackupInfoDto;
 import io.camunda.optimize.dto.optimize.rest.SnapshotInfoDto;
 import io.camunda.optimize.dto.optimize.rest.SnapshotState;
 import io.camunda.optimize.service.db.reader.BackupReader;
 import io.camunda.optimize.service.db.writer.BackupWriter;
-import io.camunda.optimize.service.exceptions.OptimizeConfigurationException;
+import io.camunda.optimize.service.exceptions.conflict.OptimizeConflictException;
 import io.camunda.optimize.service.util.configuration.ConfigurationService;
 import jakarta.ws.rs.NotFoundException;
-import java.time.Instant;
-import java.time.OffsetDateTime;
-import java.time.ZoneId;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.lang3.StringUtils;
 import org.springframework.stereotype.Component;
 
 @RequiredArgsConstructor
@@ -49,35 +43,34 @@ public class BackupService {
   private final ConfigurationService configurationService;
 
   public synchronized void triggerBackup(final Long backupId) {
-    validateRepositoryExists();
-    backupReader.validateNoDuplicateBackupId(backupId);
+    backupReader.validateRepositoryExists();
+    validateNoDuplicateBackupId(backupId);
 
     log.info("Triggering backup with ID {}", backupId);
     backupWriter.triggerSnapshotCreation(backupId);
   }
 
   public List<BackupInfoDto> getAllBackupInfo() {
-    validateRepositoryExists();
+    backupReader.validateRepositoryExists();
     return backupReader.getAllOptimizeSnapshotsByBackupId().entrySet().stream()
         .map(
             entry ->
                 getSingleBackupInfo(
                     entry.getKey(),
-                    entry.getValue().stream()
-                        .collect(groupingBy(s -> SnapshotState.valueOf(s.state())))))
+                    entry.getValue().stream().collect(groupingBy(SnapshotInfoDto::getState))))
         .toList();
   }
 
   public BackupInfoDto getSingleBackupInfo(final Long backupId) {
-    validateRepositoryExists();
+    backupReader.validateRepositoryExists();
     return getSingleBackupInfo(
         backupId,
         backupReader.getOptimizeSnapshotsForBackupId(backupId).stream()
-            .collect(groupingBy(s -> SnapshotState.valueOf(s.state()))));
+            .collect(groupingBy(SnapshotInfoDto::getState)));
   }
 
   private BackupInfoDto getSingleBackupInfo(
-      final Long backupId, final Map<SnapshotState, List<SnapshotInfo>> snapshotInfosPerState) {
+      final Long backupId, final Map<SnapshotState, List<SnapshotInfoDto>> snapshotInfosPerState) {
     if (snapshotInfosPerState.isEmpty()) {
       final String reason =
           String.format("No Optimize backup with ID [%d] could be found.", backupId);
@@ -88,7 +81,7 @@ public class BackupService {
   }
 
   private BackupInfoDto getBackupInfoDto(
-      final Long backupId, final Map<SnapshotState, List<SnapshotInfo>> snapshotInfosPerState) {
+      final Long backupId, final Map<SnapshotState, List<SnapshotInfoDto>> snapshotInfosPerState) {
     final BackupState backupState = determineBackupState(snapshotInfosPerState);
     String failureReason = null;
     if (BackupState.FAILED == backupState) {
@@ -96,36 +89,23 @@ public class BackupService {
           String.format(
               "The following snapshots failed: [%s]",
               snapshotInfosPerState.getOrDefault(FAILED, Collections.emptyList()).stream()
-                  .map(snapshotInfo -> snapshotInfo.snapshot())
+                  .map(SnapshotInfoDto::getSnapshotName)
                   .collect(joining(", ")));
     }
     return new BackupInfoDto(
         backupId,
         failureReason,
         backupState,
-        snapshotInfosPerState.values().stream()
-            .flatMap(List::stream)
-            .map(
-                snapshotInfo ->
-                    new SnapshotInfoDto(
-                        snapshotInfo.snapshot(),
-                        SnapshotState.valueOf(snapshotInfo.state()),
-                        OffsetDateTime.ofInstant(
-                            Instant.ofEpochMilli(snapshotInfo.startTime().toEpochMilli()),
-                            ZoneId.systemDefault()),
-                        snapshotInfo.shards().failures().stream()
-                            .map(ShardFailure::toString)
-                            .toList()))
-            .toList());
+        snapshotInfosPerState.values().stream().flatMap(List::stream).toList());
   }
 
   public void deleteBackup(final Long backupId) {
-    validateRepositoryExists();
+    backupReader.validateRepositoryExists();
     backupWriter.deleteOptimizeSnapshots(backupId);
   }
 
   private BackupState determineBackupState(
-      final Map<SnapshotState, List<SnapshotInfo>> snapshotInfosPerState) {
+      final Map<SnapshotState, List<SnapshotInfoDto>> snapshotInfosPerState) {
     if (snapshotInfosPerState.getOrDefault(SUCCESS, Collections.emptyList()).size()
         == EXPECTED_NUMBER_OF_SNAPSHOTS_PER_BACKUP) {
       return BackupState.COMPLETED;
@@ -146,17 +126,19 @@ public class BackupService {
     }
   }
 
-  // todo extract this validation to the readers to avoid using there configuration or extract it
-  // from db configuration by OPT-7245
-  private void validateRepositoryExists() {
-    if (StringUtils.isEmpty(
-        configurationService.getElasticSearchConfiguration().getSnapshotRepositoryName())) {
+  private void validateNoDuplicateBackupId(final Long backupId) {
+    final List<SnapshotInfoDto> existingSnapshots =
+        backupReader.getOptimizeSnapshotsForBackupId(backupId);
+    if (!existingSnapshots.isEmpty()) {
       final String reason =
-          "Cannot execute backup request because no Elasticsearch snapshot repository name found in Optimize configuration.";
+          String.format(
+              "A backup with ID [%s] already exists. Found snapshots: [%s]",
+              backupId,
+              existingSnapshots.stream()
+                  .map(SnapshotInfoDto::getSnapshotName)
+                  .collect(joining(", ")));
       log.error(reason);
-      throw new OptimizeConfigurationException(reason);
-    } else {
-      backupReader.validateRepositoryExistsOrFail();
+      throw new OptimizeConflictException(reason);
     }
   }
 }

--- a/optimize/backend/src/main/java/io/camunda/optimize/service/db/es/reader/BackupReaderES.java
+++ b/optimize/backend/src/main/java/io/camunda/optimize/service/db/es/reader/BackupReaderES.java
@@ -12,24 +12,28 @@ import static io.camunda.optimize.service.util.SnapshotUtil.SNAPSHOT_MISSING_EXC
 import static io.camunda.optimize.service.util.SnapshotUtil.getAllWildcardedSnapshotNamesForBackupId;
 import static io.camunda.optimize.service.util.SnapshotUtil.getAllWildcardedSnapshotNamesForWildcardedBackupId;
 import static java.util.stream.Collectors.groupingBy;
-import static java.util.stream.Collectors.joining;
 
 import co.elastic.clients.elasticsearch._types.ElasticsearchException;
+import co.elastic.clients.elasticsearch._types.ShardFailure;
 import co.elastic.clients.elasticsearch.snapshot.GetRepositoryRequest;
 import co.elastic.clients.elasticsearch.snapshot.GetSnapshotRequest;
 import co.elastic.clients.elasticsearch.snapshot.GetSnapshotResponse;
 import co.elastic.clients.elasticsearch.snapshot.SnapshotInfo;
+import io.camunda.optimize.dto.optimize.rest.SnapshotInfoDto;
+import io.camunda.optimize.dto.optimize.rest.SnapshotState;
 import io.camunda.optimize.service.db.es.OptimizeElasticsearchClient;
-import io.camunda.optimize.service.db.reader.BackupReader;
+import io.camunda.optimize.service.db.reader.AbstractBackupReader;
 import io.camunda.optimize.service.exceptions.OptimizeConfigurationException;
 import io.camunda.optimize.service.exceptions.OptimizeElasticsearchConnectionException;
 import io.camunda.optimize.service.exceptions.OptimizeRuntimeException;
 import io.camunda.optimize.service.exceptions.OptimizeSnapshotRepositoryNotFoundException;
-import io.camunda.optimize.service.exceptions.conflict.OptimizeConflictException;
 import io.camunda.optimize.service.util.SnapshotUtil;
 import io.camunda.optimize.service.util.configuration.ConfigurationService;
 import io.camunda.optimize.service.util.configuration.condition.ElasticSearchCondition;
 import java.io.IOException;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
@@ -44,13 +48,17 @@ import org.springframework.stereotype.Component;
 @Component
 @Slf4j
 @Conditional(ElasticSearchCondition.class)
-public class BackupReaderES implements BackupReader {
-
+public class BackupReaderES extends AbstractBackupReader {
   private final OptimizeElasticsearchClient esClient;
   private final ConfigurationService configurationService;
 
   @Override
-  public void validateRepositoryExistsOrFail() {
+  protected String getSnapshotRepositoryName() {
+    return configurationService.getElasticSearchConfiguration().getSnapshotRepositoryName();
+  }
+
+  @Override
+  protected void validateRepositoryExistsOrFail() {
     final String repositoryName = getRepositoryName();
     if (StringUtils.isEmpty(repositoryName)) {
       final String reason =
@@ -62,7 +70,7 @@ public class BackupReaderES implements BackupReader {
           GetRepositoryRequest.of(b -> b.name(repositoryName));
       try {
         esClient.verifyRepositoryExists(getRepositoriesRequest);
-      } catch (ElasticsearchException e) {
+      } catch (final ElasticsearchException e) {
         if (StringUtils.contains(e.getMessage(), REPOSITORY_MISSING_EXCEPTION_TYPE)) {
           final String reason =
               String.format("No repository with name [%s] could be found.", repositoryName);
@@ -76,7 +84,7 @@ public class BackupReaderES implements BackupReader {
           log.error(reason, e);
           throw new OptimizeRuntimeException(reason, e);
         }
-      } catch (IOException e) {
+      } catch (final IOException e) {
         final String reason =
             String.format(
                 "Encountered an error connecting to Elasticsearch while retrieving repository with name [%s].",
@@ -88,45 +96,45 @@ public class BackupReaderES implements BackupReader {
   }
 
   @Override
-  public void validateNoDuplicateBackupId(final Long backupId) {
-    final List<SnapshotInfo> existingSnapshots = getOptimizeSnapshotsForBackupId(backupId);
-    if (!existingSnapshots.isEmpty()) {
-      final String reason =
-          String.format(
-              "A backup with ID [%s] already exists. Found snapshots: [%s]",
-              backupId,
-              existingSnapshots.stream().map(SnapshotInfo::snapshot).collect(joining(", ")));
-      log.error(reason);
-      throw new OptimizeConflictException(reason);
-    }
-  }
-
-  @Override
-  public Map<Long, List<SnapshotInfo>> getAllOptimizeSnapshotsByBackupId() {
+  public Map<Long, List<SnapshotInfoDto>> getAllOptimizeSnapshotsByBackupId() {
     return getAllOptimizeSnapshots().stream()
+        .map(BackupReaderES::toSnapshotInfoDto)
         .collect(
             groupingBy(
-                snapshotInfo -> SnapshotUtil.getBackupIdFromSnapshotName(snapshotInfo.snapshot())));
+                snapshotInfo ->
+                    SnapshotUtil.getBackupIdFromSnapshotName(snapshotInfo.getSnapshotName())));
   }
 
   @Override
-  public List<SnapshotInfo> getAllOptimizeSnapshots() {
+  public List<SnapshotInfoDto> getOptimizeSnapshotsForBackupId(final Long backupId) {
+    return getOptimizeSnapshots(getAllWildcardedSnapshotNamesForBackupId(backupId)).stream()
+        .map(BackupReaderES::toSnapshotInfoDto)
+        .toList();
+  }
+
+  private static SnapshotInfoDto toSnapshotInfoDto(final SnapshotInfo snapshotInfo) {
+    final long startTimeMillis = snapshotInfo.startTimeInMillis();
+    final List<String> shardFailures =
+        snapshotInfo.shards().failures().stream().map(ShardFailure::toString).toList();
+    return new SnapshotInfoDto(
+        snapshotInfo.snapshot(),
+        SnapshotState.valueOf(snapshotInfo.state()),
+        OffsetDateTime.ofInstant(Instant.ofEpochMilli(startTimeMillis), ZoneId.systemDefault()),
+        shardFailures);
+  }
+
+  private List<SnapshotInfo> getAllOptimizeSnapshots() {
     return getOptimizeSnapshots(getAllWildcardedSnapshotNamesForWildcardedBackupId());
-  }
-
-  @Override
-  public List<SnapshotInfo> getOptimizeSnapshotsForBackupId(final Long backupId) {
-    return getOptimizeSnapshots(getAllWildcardedSnapshotNamesForBackupId(backupId));
   }
 
   private List<SnapshotInfo> getOptimizeSnapshots(final String[] snapshots) {
     final GetSnapshotRequest snapshotsStatusRequest =
         GetSnapshotRequest.of(
             b -> b.repository(getRepositoryName()).snapshot(Arrays.stream(snapshots).toList()));
-    GetSnapshotResponse response;
+    final GetSnapshotResponse response;
     try {
       response = esClient.getSnapshots(snapshotsStatusRequest);
-    } catch (ElasticsearchException e) {
+    } catch (final ElasticsearchException e) {
       if (StringUtils.contains(e.getMessage(), SNAPSHOT_MISSING_EXCEPTION_TYPE)) {
         // no snapshot with given backupID exists
         return Collections.emptyList();
@@ -137,7 +145,7 @@ public class BackupReaderES implements BackupReader {
               String.join(", ", snapshots));
       log.error(reason);
       throw new OptimizeRuntimeException(reason, e);
-    } catch (IOException e) {
+    } catch (final IOException e) {
       final String reason =
           String.format(
               "Encountered an error connecting to Elasticsearch while retrieving snapshots with names [%s].",

--- a/optimize/backend/src/main/java/io/camunda/optimize/service/db/os/reader/BackupReaderOS.java
+++ b/optimize/backend/src/main/java/io/camunda/optimize/service/db/os/reader/BackupReaderOS.java
@@ -62,7 +62,7 @@ public class BackupReaderOS extends AbstractBackupReader {
     final String repositoryName = getRepositoryName();
     if (StringUtils.isEmpty(repositoryName)) {
       final String reason =
-          "Cannot trigger backup because no Opensearch snapshot repository name found in Optimize configuration.";
+          "No Opensearch snapshot repository name found in Optimize configuration.";
       log.error(reason);
       throw new OptimizeConfigurationException(reason);
     } else {
@@ -70,7 +70,7 @@ public class BackupReaderOS extends AbstractBackupReader {
           GetRepositoryRequest.of(b -> b.name(repositoryName));
       try {
         osClient.verifyRepositoryExists(getRepositoriesRequest);
-      } catch (OpenSearchException e) {
+      } catch (final OpenSearchException e) {
         if (StringUtils.contains(e.getMessage(), REPOSITORY_MISSING_EXCEPTION_TYPE)) {
           final String reason =
               String.format("No repository with name [%s] could be found.", repositoryName);
@@ -84,7 +84,7 @@ public class BackupReaderOS extends AbstractBackupReader {
           log.error(reason, e);
           throw new OptimizeRuntimeException(reason, e);
         }
-      } catch (IOException e) {
+      } catch (final IOException e) {
         final String reason =
             String.format(
                 "Encountered an error connecting to OpenSearch while retrieving repository with name [%s].",
@@ -131,10 +131,10 @@ public class BackupReaderOS extends AbstractBackupReader {
     final GetSnapshotRequest snapshotsStatusRequest =
         GetSnapshotRequest.of(
             b -> b.repository(getRepositoryName()).snapshot(Arrays.stream(snapshots).toList()));
-    GetSnapshotResponse response;
+    final GetSnapshotResponse response;
     try {
       response = osClient.getSnapshots(snapshotsStatusRequest);
-    } catch (OpenSearchException e) {
+    } catch (final OpenSearchException e) {
       if (StringUtils.contains(e.getMessage(), SNAPSHOT_MISSING_EXCEPTION_TYPE)) {
         // no snapshot with given backupID exists
         return Collections.emptyList();
@@ -145,7 +145,7 @@ public class BackupReaderOS extends AbstractBackupReader {
               String.join(", ", snapshots));
       log.error(reason);
       throw new OptimizeRuntimeException(reason, e);
-    } catch (IOException e) {
+    } catch (final IOException e) {
       final String reason =
           String.format(
               "Encountered an error connecting to OpenSearch while retrieving snapshots with names [%s].",

--- a/optimize/backend/src/main/java/io/camunda/optimize/service/db/os/reader/BackupReaderOS.java
+++ b/optimize/backend/src/main/java/io/camunda/optimize/service/db/os/reader/BackupReaderOS.java
@@ -7,14 +7,40 @@
  */
 package io.camunda.optimize.service.db.os.reader;
 
-import co.elastic.clients.elasticsearch.snapshot.SnapshotInfo;
-import io.camunda.optimize.service.db.reader.BackupReader;
+import static io.camunda.optimize.service.util.SnapshotUtil.REPOSITORY_MISSING_EXCEPTION_TYPE;
+import static io.camunda.optimize.service.util.SnapshotUtil.SNAPSHOT_MISSING_EXCEPTION_TYPE;
+import static io.camunda.optimize.service.util.SnapshotUtil.getAllWildcardedSnapshotNamesForBackupId;
+import static io.camunda.optimize.service.util.SnapshotUtil.getAllWildcardedSnapshotNamesForWildcardedBackupId;
+import static java.util.stream.Collectors.groupingBy;
+
+import io.camunda.optimize.dto.optimize.rest.SnapshotInfoDto;
+import io.camunda.optimize.dto.optimize.rest.SnapshotState;
+import io.camunda.optimize.service.db.os.OptimizeOpenSearchClient;
+import io.camunda.optimize.service.db.reader.AbstractBackupReader;
+import io.camunda.optimize.service.exceptions.OptimizeConfigurationException;
+import io.camunda.optimize.service.exceptions.OptimizeOpenSearchConnectionException;
+import io.camunda.optimize.service.exceptions.OptimizeRuntimeException;
+import io.camunda.optimize.service.exceptions.OptimizeSnapshotRepositoryNotFoundException;
+import io.camunda.optimize.service.util.SnapshotUtil;
+import io.camunda.optimize.service.util.configuration.ConfigurationService;
 import io.camunda.optimize.service.util.configuration.condition.OpenSearchCondition;
-import java.util.ArrayList;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.client.opensearch._types.OpenSearchException;
+import org.opensearch.client.opensearch._types.ShardFailure;
+import org.opensearch.client.opensearch.snapshot.GetRepositoryRequest;
+import org.opensearch.client.opensearch.snapshot.GetSnapshotRequest;
+import org.opensearch.client.opensearch.snapshot.GetSnapshotResponse;
+import org.opensearch.client.opensearch.snapshot.SnapshotInfo;
 import org.springframework.context.annotation.Conditional;
 import org.springframework.stereotype.Component;
 
@@ -22,33 +48,115 @@ import org.springframework.stereotype.Component;
 @Component
 @Slf4j
 @Conditional(OpenSearchCondition.class)
-public class BackupReaderOS implements BackupReader {
+public class BackupReaderOS extends AbstractBackupReader {
+  private final ConfigurationService configurationService;
+  private final OptimizeOpenSearchClient osClient;
 
   @Override
-  public void validateRepositoryExistsOrFail() {
-    log.debug("Functionality not implemented for OpenSearch");
+  protected String getSnapshotRepositoryName() {
+    return configurationService.getOpenSearchConfiguration().getSnapshotRepositoryName();
   }
 
   @Override
-  public void validateNoDuplicateBackupId(final Long backupId) {
-    log.debug("Functionality not implemented for OpenSearch");
+  protected void validateRepositoryExistsOrFail() {
+    final String repositoryName = getRepositoryName();
+    if (StringUtils.isEmpty(repositoryName)) {
+      final String reason =
+          "Cannot trigger backup because no Opensearch snapshot repository name found in Optimize configuration.";
+      log.error(reason);
+      throw new OptimizeConfigurationException(reason);
+    } else {
+      final GetRepositoryRequest getRepositoriesRequest =
+          GetRepositoryRequest.of(b -> b.name(repositoryName));
+      try {
+        osClient.verifyRepositoryExists(getRepositoriesRequest);
+      } catch (OpenSearchException e) {
+        if (StringUtils.contains(e.getMessage(), REPOSITORY_MISSING_EXCEPTION_TYPE)) {
+          final String reason =
+              String.format("No repository with name [%s] could be found.", repositoryName);
+          log.error(reason, e);
+          throw new OptimizeSnapshotRepositoryNotFoundException(reason, e);
+        } else {
+          final String reason =
+              String.format(
+                  "Error while retrieving repository with name [%s] due to an OpenSearchException.",
+                  repositoryName);
+          log.error(reason, e);
+          throw new OptimizeRuntimeException(reason, e);
+        }
+      } catch (IOException e) {
+        final String reason =
+            String.format(
+                "Encountered an error connecting to OpenSearch while retrieving repository with name [%s].",
+                repositoryName);
+        log.error(reason, e);
+        throw new OptimizeOpenSearchConnectionException(reason, e);
+      }
+    }
   }
 
   @Override
-  public Map<Long, List<SnapshotInfo>> getAllOptimizeSnapshotsByBackupId() {
-    log.debug("Functionality not implemented for OpenSearch");
-    return null;
+  public Map<Long, List<SnapshotInfoDto>> getAllOptimizeSnapshotsByBackupId() {
+    return getAllOptimizeSnapshots().stream()
+        .map(BackupReaderOS::toSnapshotInfoDto)
+        .collect(
+            groupingBy(
+                snapshotInfo ->
+                    SnapshotUtil.getBackupIdFromSnapshotName(snapshotInfo.getSnapshotName())));
   }
 
   @Override
-  public List<SnapshotInfo> getAllOptimizeSnapshots() {
-    log.debug("Functionality not implemented for OpenSearch");
-    return new ArrayList<>();
+  public List<SnapshotInfoDto> getOptimizeSnapshotsForBackupId(final Long backupId) {
+    return getOptimizeSnapshots(getAllWildcardedSnapshotNamesForBackupId(backupId)).stream()
+        .map(BackupReaderOS::toSnapshotInfoDto)
+        .toList();
   }
 
-  @Override
-  public List<SnapshotInfo> getOptimizeSnapshotsForBackupId(final Long backupId) {
-    log.debug("Functionality not implemented for OpenSearch");
-    return new ArrayList<>();
+  private static SnapshotInfoDto toSnapshotInfoDto(final SnapshotInfo snapshotInfo) {
+    final long startTimeMillis = Long.parseLong(snapshotInfo.startTimeInMillis());
+    final List<String> shardFailures =
+        snapshotInfo.shards().failures().stream().map(ShardFailure::toString).toList();
+    return new SnapshotInfoDto(
+        snapshotInfo.snapshot(),
+        SnapshotState.valueOf(snapshotInfo.state()),
+        OffsetDateTime.ofInstant(Instant.ofEpochMilli(startTimeMillis), ZoneId.systemDefault()),
+        shardFailures);
+  }
+
+  private List<SnapshotInfo> getAllOptimizeSnapshots() {
+    return getOptimizeSnapshots(getAllWildcardedSnapshotNamesForWildcardedBackupId());
+  }
+
+  private List<SnapshotInfo> getOptimizeSnapshots(final String[] snapshots) {
+    final GetSnapshotRequest snapshotsStatusRequest =
+        GetSnapshotRequest.of(
+            b -> b.repository(getRepositoryName()).snapshot(Arrays.stream(snapshots).toList()));
+    GetSnapshotResponse response;
+    try {
+      response = osClient.getSnapshots(snapshotsStatusRequest);
+    } catch (OpenSearchException e) {
+      if (StringUtils.contains(e.getMessage(), SNAPSHOT_MISSING_EXCEPTION_TYPE)) {
+        // no snapshot with given backupID exists
+        return Collections.emptyList();
+      }
+      final String reason =
+          String.format(
+              "Could not retrieve snapshots with names [%s] due to an OpenSearchException.",
+              String.join(", ", snapshots));
+      log.error(reason);
+      throw new OptimizeRuntimeException(reason, e);
+    } catch (IOException e) {
+      final String reason =
+          String.format(
+              "Encountered an error connecting to OpenSearch while retrieving snapshots with names [%s].",
+              String.join(", ", snapshots));
+      log.error(reason, e);
+      throw new OptimizeOpenSearchConnectionException(reason, e);
+    }
+    return response.snapshots();
+  }
+
+  private String getRepositoryName() {
+    return configurationService.getOpenSearchConfiguration().getSnapshotRepositoryName();
   }
 }

--- a/optimize/backend/src/main/java/io/camunda/optimize/service/db/reader/AbstractBackupReader.java
+++ b/optimize/backend/src/main/java/io/camunda/optimize/service/db/reader/AbstractBackupReader.java
@@ -1,0 +1,31 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.service.db.reader;
+
+import io.camunda.optimize.service.exceptions.OptimizeConfigurationException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+@Slf4j
+public abstract class AbstractBackupReader implements BackupReader {
+  @Override
+  public void validateRepositoryExists() {
+    if (StringUtils.isEmpty(getSnapshotRepositoryName())) {
+      final String reason =
+          "Cannot execute backup request because no snapshot repository name found in Optimize configuration.";
+      log.error(reason);
+      throw new OptimizeConfigurationException(reason);
+    } else {
+      validateRepositoryExistsOrFail();
+    }
+  }
+
+  protected abstract String getSnapshotRepositoryName();
+
+  protected abstract void validateRepositoryExistsOrFail();
+}

--- a/optimize/backend/src/main/java/io/camunda/optimize/service/db/reader/BackupReader.java
+++ b/optimize/backend/src/main/java/io/camunda/optimize/service/db/reader/BackupReader.java
@@ -7,19 +7,14 @@
  */
 package io.camunda.optimize.service.db.reader;
 
-import co.elastic.clients.elasticsearch.snapshot.SnapshotInfo;
+import io.camunda.optimize.dto.optimize.rest.SnapshotInfoDto;
 import java.util.List;
 import java.util.Map;
 
 public interface BackupReader {
+  void validateRepositoryExists();
 
-  void validateRepositoryExistsOrFail();
+  Map<Long, List<SnapshotInfoDto>> getAllOptimizeSnapshotsByBackupId();
 
-  void validateNoDuplicateBackupId(final Long backupId);
-
-  Map<Long, List<SnapshotInfo>> getAllOptimizeSnapshotsByBackupId();
-
-  List<SnapshotInfo> getAllOptimizeSnapshots();
-
-  List<SnapshotInfo> getOptimizeSnapshotsForBackupId(final Long backupId);
+  List<SnapshotInfoDto> getOptimizeSnapshotsForBackupId(final Long backupId);
 }

--- a/optimize/backend/src/main/java/io/camunda/optimize/service/exceptions/OptimizeOpenSearchConnectionException.java
+++ b/optimize/backend/src/main/java/io/camunda/optimize/service/exceptions/OptimizeOpenSearchConnectionException.java
@@ -1,0 +1,21 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.service.exceptions;
+
+public class OptimizeOpenSearchConnectionException extends OptimizeRuntimeException {
+  public static final String ERROR_CODE = "openSearchConnectionError";
+
+  public OptimizeOpenSearchConnectionException(final String message, final Throwable e) {
+    super(message, e);
+  }
+
+  @Override
+  public String getErrorCode() {
+    return ERROR_CODE;
+  }
+}

--- a/optimize/backend/src/test/java/io/camunda/optimize/service/backup/BackupServiceTest.java
+++ b/optimize/backend/src/test/java/io/camunda/optimize/service/backup/BackupServiceTest.java
@@ -12,9 +12,12 @@ import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+import io.camunda.optimize.dto.optimize.rest.SnapshotInfoDto;
 import io.camunda.optimize.service.BackupService;
 import io.camunda.optimize.service.db.reader.BackupReader;
 import io.camunda.optimize.service.db.writer.BackupWriter;
@@ -26,11 +29,11 @@ import io.camunda.optimize.service.util.configuration.ConfigurationService;
 import io.camunda.optimize.service.util.configuration.ElasticSearchConfiguration;
 import jakarta.ws.rs.NotFoundException;
 import java.util.Collections;
+import java.util.List;
 import lombok.SneakyThrows;
 import org.apache.commons.lang3.StringUtils;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
@@ -42,25 +45,16 @@ import org.mockito.junit.jupiter.MockitoExtension;
 @ExtendWith(MockitoExtension.class)
 public class BackupServiceTest {
 
-  @Mock private BackupReader backupReader;
-
-  @Mock private BackupWriter backupWriter;
-
-  @Mock private ElasticSearchConfiguration databaseConfiguration;
-
-  @Mock private ConfigurationService configurationService;
-
-  @InjectMocks private BackupService backupService;
   private static MockedStatic<StringUtils> stringUtils;
+  @Mock private BackupReader backupReader;
+  @Mock private BackupWriter backupWriter;
+  @Mock private ElasticSearchConfiguration databaseConfiguration;
+  @Mock private ConfigurationService configurationService;
+  @InjectMocks private BackupService backupService;
 
   @BeforeAll
   public static void beforeAll() {
     stringUtils = Mockito.mockStatic(StringUtils.class);
-  }
-
-  @BeforeEach
-  public void beforeEach() {
-    when(configurationService.getElasticSearchConfiguration()).thenReturn(databaseConfiguration);
   }
 
   @AfterAll
@@ -71,13 +65,17 @@ public class BackupServiceTest {
   @Test
   public void triggerBackupWithoutSnapshotRepositoryConfig() {
     // given mock no repository in config
-    stringUtils.when(() -> StringUtils.isEmpty(any())).thenReturn(true);
+    final OptimizeConfigurationException optimizeConfigurationException =
+        new OptimizeConfigurationException(
+            "Cannot execute backup request because no snapshot repository name found in Optimize configuration.");
+    doThrow(optimizeConfigurationException).when(backupReader).validateRepositoryExists();
+
     // when/then
     final OptimizeConfigurationException thrown =
         assertThrows(OptimizeConfigurationException.class, () -> backupService.triggerBackup(123L));
     assertThat(thrown.getMessage())
         .isEqualTo(
-            "Cannot execute backup request because no Elasticsearch snapshot repository name found in Optimize "
+            "Cannot execute backup request because no snapshot repository name found in Optimize "
                 + "configuration.");
   }
 
@@ -90,7 +88,7 @@ public class BackupServiceTest {
             new OptimizeSnapshotRepositoryNotFoundException(
                 "No repository with name [does_not_exist] could be found."))
         .when(backupReader)
-        .validateRepositoryExistsOrFail();
+        .validateRepositoryExists();
 
     // when/then
     final OptimizeRuntimeException thrown =
@@ -105,13 +103,13 @@ public class BackupServiceTest {
     // given
     // Mock existence of repository name field in config
     stringUtils.when(() -> StringUtils.isEmpty(any())).thenReturn(false);
-    doNothing().when(backupReader).validateRepositoryExistsOrFail();
     // Mock existence of other backup with the same ID
-    doThrow(
-            new OptimizeConflictException(
-                "A backup with ID [123] already exists. Found snapshots: [existingSnapshotName/Xtll5DxHQ56j6rMz8nFDmQ]"))
+    final SnapshotInfoDto snapshotInfoDtoMock = mock(SnapshotInfoDto.class);
+    when(snapshotInfoDtoMock.getSnapshotName())
+        .thenReturn("existingSnapshotName/Xtll5DxHQ56j6rMz8nFDmQ");
+    doReturn(List.of(snapshotInfoDtoMock))
         .when(backupReader)
-        .validateNoDuplicateBackupId(any());
+        .getOptimizeSnapshotsForBackupId(any());
 
     // when
     final OptimizeConflictException thrown =
@@ -125,7 +123,10 @@ public class BackupServiceTest {
   @Test
   public void getSpecificBackupStateWithoutSnapshotRepositoryConfig() {
     // given mock no repository in config
-    stringUtils.when(() -> StringUtils.isEmpty(any())).thenReturn(true);
+    final OptimizeConfigurationException optimizeConfigurationException =
+        new OptimizeConfigurationException(
+            "Cannot execute backup request because no snapshot repository name found in Optimize configuration.");
+    doThrow(optimizeConfigurationException).when(backupReader).validateRepositoryExists();
 
     // when/then
     final OptimizeConfigurationException thrown =
@@ -133,7 +134,7 @@ public class BackupServiceTest {
             OptimizeConfigurationException.class, () -> backupService.getSingleBackupInfo(123L));
     assertThat(thrown.getMessage())
         .isEqualTo(
-            "Cannot execute backup request because no Elasticsearch snapshot repository name found in Optimize "
+            "Cannot execute backup request because no snapshot repository name found in Optimize "
                 + "configuration.");
   }
 
@@ -146,7 +147,7 @@ public class BackupServiceTest {
             new OptimizeSnapshotRepositoryNotFoundException(
                 "No repository with name [does_not_exist] could be found."))
         .when(backupReader)
-        .validateRepositoryExistsOrFail();
+        .validateRepositoryExists();
 
     // when/then
     final OptimizeRuntimeException thrown =
@@ -160,7 +161,7 @@ public class BackupServiceTest {
     // given
     // Mock existence of repository name field in config
     stringUtils.when(() -> StringUtils.isEmpty(any())).thenReturn(false);
-    doNothing().when(backupReader).validateRepositoryExistsOrFail();
+    doNothing().when(backupReader).validateRepositoryExists();
     when(backupReader.getOptimizeSnapshotsForBackupId(anyLong()))
         .thenReturn(Collections.emptyList());
 
@@ -173,14 +174,17 @@ public class BackupServiceTest {
   @Test
   public void getAllBackupStateWithoutSnapshotRepositoryConfig() {
     // given mock no repository in config
-    stringUtils.when(() -> StringUtils.isEmpty(any())).thenReturn(true);
+    final OptimizeConfigurationException optimizeConfigurationException =
+        new OptimizeConfigurationException(
+            "Cannot execute backup request because no snapshot repository name found in Optimize configuration.");
+    doThrow(optimizeConfigurationException).when(backupReader).validateRepositoryExists();
 
     // when/then
     final OptimizeConfigurationException thrown =
         assertThrows(OptimizeConfigurationException.class, () -> backupService.getAllBackupInfo());
     assertThat(thrown.getMessage())
         .isEqualTo(
-            "Cannot execute backup request because no Elasticsearch snapshot repository name found in Optimize "
+            "Cannot execute backup request because no snapshot repository name found in Optimize "
                 + "configuration.");
   }
 
@@ -193,7 +197,7 @@ public class BackupServiceTest {
             new OptimizeSnapshotRepositoryNotFoundException(
                 "No repository with name [does_not_exist] could be found."))
         .when(backupReader)
-        .validateRepositoryExistsOrFail();
+        .validateRepositoryExists();
 
     // when/then
     final OptimizeRuntimeException thrown =
@@ -207,7 +211,7 @@ public class BackupServiceTest {
     // given
     // Mock existence of repository name field in config
     stringUtils.when(() -> StringUtils.isEmpty(any())).thenReturn(false);
-    doNothing().when(backupReader).validateRepositoryExistsOrFail();
+    doNothing().when(backupReader).validateRepositoryExists();
     when(backupReader.getAllOptimizeSnapshotsByBackupId()).thenReturn(Collections.emptyMap());
 
     // when/then
@@ -217,14 +221,17 @@ public class BackupServiceTest {
   @Test
   public void deleteBackupWithoutSnapshotRepositoryConfig() {
     // given mock no repository in config
-    stringUtils.when(() -> StringUtils.isEmpty(any())).thenReturn(true);
+    final OptimizeConfigurationException optimizeConfigurationException =
+        new OptimizeConfigurationException(
+            "Cannot execute backup request because no snapshot repository name found in Optimize configuration.");
+    doThrow(optimizeConfigurationException).when(backupReader).validateRepositoryExists();
 
     // when/then
     final OptimizeConfigurationException thrown =
         assertThrows(OptimizeConfigurationException.class, () -> backupService.deleteBackup(123L));
     assertThat(thrown.getMessage())
         .isEqualTo(
-            "Cannot execute backup request because no Elasticsearch snapshot repository name found in Optimize "
+            "Cannot execute backup request because no snapshot repository name found in Optimize "
                 + "configuration.");
   }
 
@@ -237,7 +244,7 @@ public class BackupServiceTest {
             new OptimizeSnapshotRepositoryNotFoundException(
                 "No repository with name [does_not_exist] could be found."))
         .when(backupReader)
-        .validateRepositoryExistsOrFail();
+        .validateRepositoryExists();
 
     // when/then
     final OptimizeRuntimeException thrown =

--- a/optimize/util/optimize-commons/src/main/java/io/camunda/optimize/service/db/os/ExtendedOpenSearchClient.java
+++ b/optimize/util/optimize-commons/src/main/java/io/camunda/optimize/service/db/os/ExtendedOpenSearchClient.java
@@ -14,8 +14,11 @@ import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import jakarta.json.stream.JsonGenerator;
+import jakarta.json.stream.JsonParser;
+import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
@@ -24,6 +27,8 @@ import java.util.regex.Pattern;
 import org.opensearch.client.Request;
 import org.opensearch.client.Response;
 import org.opensearch.client.json.JsonpDeserializer;
+import org.opensearch.client.json.JsonpMapper;
+import org.opensearch.client.json.ObjectBuilderDeserializer;
 import org.opensearch.client.json.jackson.JacksonJsonpMapper;
 import org.opensearch.client.opensearch.OpenSearchClient;
 import org.opensearch.client.opensearch._types.ErrorResponse;
@@ -31,6 +36,10 @@ import org.opensearch.client.opensearch._types.OpenSearchException;
 import org.opensearch.client.opensearch.core.CountResponse;
 import org.opensearch.client.opensearch.core.SearchRequest;
 import org.opensearch.client.opensearch.core.SearchResponse;
+import org.opensearch.client.opensearch.snapshot.GetSnapshotRequest;
+import org.opensearch.client.opensearch.snapshot.GetSnapshotResponse;
+import org.opensearch.client.opensearch.snapshot.GetSnapshotResponse.Builder;
+import org.opensearch.client.opensearch.snapshot.SnapshotInfo;
 import org.opensearch.client.transport.JsonEndpoint;
 import org.opensearch.client.transport.OpenSearchTransport;
 import org.opensearch.client.transport.endpoints.EndpointWithResponseMapperAttr;
@@ -132,6 +141,37 @@ public class ExtendedOpenSearchClient extends OpenSearchClient {
     return arbitraryRequest(json, endpoint);
   }
 
+  /**
+   * Standard opensearch GetSnapshotResponse builder considers fields "total" and "remaining" to be
+   * mandatory in response. However, OS server doesn't provide them, so workarounding the
+   * getSnapshots request by setting them to 0.
+   *
+   * @param getSnapshotRequest
+   * @return
+   * @throws IOException
+   * @throws OpenSearchException
+   */
+  public GetSnapshotResponse getSnapshots(final GetSnapshotRequest getSnapshotRequest)
+      throws IOException, OpenSearchException {
+    final JsonpMapper jsonpMapper = transport.jsonpMapper();
+    final String snapshots = String.join(",", getSnapshotRequest.snapshot());
+    final JsonpDeserializer<GetSnapshotResponse> deserializer =
+        ObjectBuilderDeserializer.lazy(
+            () -> new Builder().total(0).remaining(0),
+            op ->
+                op.add(
+                    Builder::snapshots,
+                    JsonpDeserializer.arrayDeserializer(SnapshotInfo._DESERIALIZER),
+                    "snapshots"));
+    final String json =
+        arbitraryRequestAsString(
+            "GET", format("/_snapshot/%s/%s", getSnapshotRequest.repository(), snapshots), "{}");
+    final InputStream is = new ByteArrayInputStream(json.getBytes());
+    try (final JsonParser parser = jsonpMapper.jsonProvider().createParser(is)) {
+      return deserializer.deserialize(parser, jsonpMapper);
+    }
+  }
+
   private <R> R arbitraryRequest(
       final String json, final JsonEndpoint<Map<String, Object>, R, ErrorResponse> endpoint)
       throws IOException, OpenSearchException {
@@ -148,7 +188,7 @@ public class ExtendedOpenSearchClient extends OpenSearchClient {
                 e -> {
                   try {
                     return jsonToMap(new String(e.getContent().readAllBytes()));
-                  } catch (IOException ex) {
+                  } catch (final IOException ex) {
                     return new HashMap<String, Object>();
                   }
                 })

--- a/optimize/util/optimize-commons/src/main/java/io/camunda/optimize/service/db/os/OptimizeOpenSearchClient.java
+++ b/optimize/util/optimize-commons/src/main/java/io/camunda/optimize/service/db/os/OptimizeOpenSearchClient.java
@@ -43,6 +43,7 @@ import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.CompletableFuture;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Supplier;
@@ -106,6 +107,11 @@ import org.opensearch.client.opensearch.indices.GetMappingResponse;
 import org.opensearch.client.opensearch.indices.RolloverRequest;
 import org.opensearch.client.opensearch.indices.RolloverResponse;
 import org.opensearch.client.opensearch.indices.rollover.RolloverConditions;
+import org.opensearch.client.opensearch.snapshot.CreateSnapshotRequest;
+import org.opensearch.client.opensearch.snapshot.CreateSnapshotResponse;
+import org.opensearch.client.opensearch.snapshot.GetRepositoryRequest;
+import org.opensearch.client.opensearch.snapshot.GetSnapshotRequest;
+import org.opensearch.client.opensearch.snapshot.GetSnapshotResponse;
 import org.opensearch.client.opensearch.tasks.GetTasksResponse;
 import org.opensearch.client.opensearch.tasks.ListRequest;
 import org.opensearch.client.opensearch.tasks.ListResponse;
@@ -1025,5 +1031,23 @@ public class OptimizeOpenSearchClient extends DatabaseClient {
             format("Error while trying to read Opensearch task (ID: %s) progress!", taskId), e);
       }
     }
+  }
+
+  public void verifyRepositoryExists(final GetRepositoryRequest getRepositoriesRequest)
+      throws IOException, OpenSearchException {
+    openSearchClient.snapshot().getRepository(getRepositoriesRequest);
+  }
+
+  public GetSnapshotResponse getSnapshots(final GetSnapshotRequest getSnapshotRequest)
+      throws IOException {
+    return openSearchClient.getSnapshots(getSnapshotRequest);
+  }
+
+  public CompletableFuture<CreateSnapshotResponse> triggerSnapshotAsync(
+      final CreateSnapshotRequest createSnapshotRequest) {
+    return safe(
+        () -> openSearchAsyncClient.snapshot().create(createSnapshotRequest),
+        e -> "Failed to triger snapshot creation!",
+        log);
   }
 }

